scripts/analyze

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import json
+import re
+import shutil
+
+force = ['']
+featurefile = ''
+
+for i in sys.argv:
+    if i == sys.argv[0]:
+        continue
+    option = i.split('=')[0]
+    value = i.split('=')[1]
+    if option == 'BOARD':
+        board = value
+    elif option == 'SCRIPT':
+        script = value
+    elif option == 'FORCE':
+        force = value.split(',')
+    elif option == 'PRJBASE':
+        prjbase = value
+    elif option == 'PRJCONF':
+        prjconf = value
+    elif option == 'MAKEFILE':
+        makefile = value
+    elif option == 'MAKEBASE':
+        makebase = value
+    elif option == 'PROFILE':
+        featurefile = value
+
+print("Analyzing %s for %s" % (script, board))
+
+dependencies = []
+zephyr_config = []
+src_list = []
+zjs_config = []
+jrs_config = []
+
+def write_jrsconfig(list):
+    if featurefile != '':
+        if os.path.exists(featurefile):
+            shutil.copyfile(featurefile, featurefile + '.bak')
+        with open("fragments/jerry_feature.base", "r") as jrs:
+            feature = jrs.read()
+        with open(featurefile, "w") as p:
+            for i in list:
+                feature = feature.replace(i, "#" + i)
+            p.write(feature)
+
+def write_makefile(list, file):
+    with open(makebase, "r") as b:
+        base = b.read()
+    with open(file, "w") as f:
+        f.write(base)
+        if list:
+            f.write("obj-y += \\\n")
+            for i in list:
+                f.write("\t" + i.replace(".c", ".o") + " \\\n")
+            f.write("\nccflags-y += ")
+            for i in zjs_config:
+                f.write(i + " ")
+            f.write("\n")
+
+def write_zconf(list, file):
+    with open(file, "w") as f:
+        for i in list:
+            f.write(i + '\n')
+
+def add_to_list(list1, list2):
+    if list1 == None:
+        return list2
+    if list2 == None:
+        return list1
+    list1_set = set(list1)
+    list2_set = set(list2)
+    diff = list2_set - list1_set
+    return list1 + list(diff)
```

Comment:
You might be doing this to preserve, order but I'm not sure you need to do that. You could replace it with this:
`return list(set(list1) | set(list2))`

And you might not even really need to convert to list, it's still iterable.